src/lib/operators/abstract_read_write_operator.cpp

```diff
@@ -17,16 +17,7 @@ void AbstractReadWriteOperator::execute() {
 
   Assert(_state == ReadWriteOperatorState::Pending, "Operator needs to have state Pending in order to be executed.");
 
-  try {
-    AbstractOperator::execute();
-  } catch (...) {
-    // No matter what goes wrong, we need to mark the operators as failed. Otherwise, when the transaction context
-    // gets destroyed, it will cause another exception that hides the one that caused the actual error. We are NOT
-    // trying to handle the exception here - just making sure that we are not misled when we debug things.
-    _mark_as_failed();
```

Comment:
`_mark_as_failed` is there to mark MVCC-conflicts, NOT to handle arbitrary exception thrown from Asserts.

Reply:
This is not about "handling" exceptions, it is about preventing MVCC sanity checks to fail during stack unwinding. Those checks would hide the original error.

Reply:
I don't understand, can you elaborate?

(From what was happening to me, any Assert/exception thrown in a ReadWrite Operator would be caught here, its (useful) error message would be discarded and the destructor of the TransactionContext would trigger a terminate because "a failed operator was not rolled back")

Reply:
sure, but again, why would you trigger the "keep hyrise alive" MVCC mechanism "_mark_as_failed()" when an Assert exception has been thrown, which is our way of killing hyrise?

Reply:
Exceptions and MVCC conflicts are two entirely different things and I feel like they got mixed up here.

Reply:
So far, I am unsuccessful in reproducing the problems that led to this. We can change it as proposed, but I am somewhat sure that we will have to come back to this.

Reply:
> The original error message should be rethrown by throw below, shouldn't it?

I think the reason we're not seeing error messages from RW operators is that the exception from the destructor of the TransactionContext terminates the program before the exception from the RW operator reaches the top of the stack.

Reply:
Okay, @mrks, I believe I know now why the try/catch block that I removed was there. If I am right, it is still an ugly hack and we need to solve the underlying problem better.

1. So, we want to test that inserting NULL values into a non-nullable column throws an exception, i.e., Hyrise can detect that an invalid operation is performed and crash gracefully instead of segfaulting or something. All nice and well.
2. For `Insert` to even reach the `Assert` we want to test, it needs a `TransactionContext`. So we create one and assign it to the `Insert`.
3. Since `Insert` throws an  exception in the middle of its operation, it remains in `_state == ReadWriteOperatorState::Pending`.
4. Now we have a problem, because a `TransactionContext` cannot be destroyed legally without all its operators being either rolled back or committed. But the Insert is still in state "Pending", it can neither be commit nor rolled back, since it is neither in state "Executed" nor "Failed".
5. The deleted block to the rescue: Simply mark the Operator as mvcc-failed and it becomes legal for the `TransactionContext` to call `AbstractReadWriteOperator::rollback_records()`. Notice how we treat an object that is corrupt as far as we are concerned (since it threw an exception) to continue operating.

Reply:
For now, I added the try/catch block back in, since it is really not the point of this PR to fix this. Maybe we can discuss the on Wednesday

Reply:
Yes, that is pretty much what happened.